components/schema-migrator/migrations/director/20220203161810_ord_internal_visibility.up.sql

```diff
@@ -0,0 +1,214 @@
+BEGIN;
+
+ALTER TABLE bundle_references
+ADD COLUMN visibility varchar;
```

Comment:
Filtering happens only on the visibility column of the bundle_references and not on the visibility column of the apis/events. This is fine since the visibility column of bundle_refs is always a copy of the visibility column of the referenced api/event.

However, we can potentially break this relation in the code while introducing new features (create bundle ref and forget it).

We can discuss in the team whether this is a good candidate for a trigger/cascade and make it "automatically" equal to the visibility of the referenced API/event or we will rely on tests to cover such errors.

Another option which probably I like the most is to use a subquery in the code for the visibility check.

Something like:

`WHERE (SELECT visibility FROM apis WHERE apis.id = br.api_def_id) = 'public'`

Reply:
After a brief offline discussion, the approach with the subquery was chosen.